conformance/tests/basic/inferencepool_resolvedrefs_condition.go

```diff
@@ -0,0 +1,127 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package basic
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+
+	"sigs.k8s.io/gateway-api-inference-extension/conformance/tests"
+	k8sutils "sigs.k8s.io/gateway-api-inference-extension/conformance/utils/kubernetes"
+	gatewayk8sutils "sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+)
+
+func init() {
+	tests.ConformanceTests = append(tests.ConformanceTests, InferencePoolParentStatus)
+}
+
+var InferencePoolParentStatus = suite.ConformanceTest{
+	ShortName:   "InferencePoolResolvedRefsCondition",
+	Description: "Verify that an InferencePool correctly updates its parent-specific status (e.g., Accepted condition) when referenced by HTTPRoutes attached to shared Gateways, and clears parent statuses when no longer referenced.",
+	Manifests:   []string{"tests/basic/inferencepool_resolvedrefs_condition.yaml"},
+	Features:    []features.FeatureName{},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+		const (
+			appBackendNamespace        = "gateway-conformance-app-backend"
+			infraNamespace             = "gateway-conformance-infra"
+			poolName                   = "multi-gateway-pool"
+			sharedGateway1Name         = "conformance-gateway"
+			sharedGateway2Name         = "conformance-secondary-gateway"
+			httpRoute1Name             = "httproute-for-gw1"
+			httpRoute2Name             = "httproute-for-gw2"
+			reasonPoolAcceptedByParent = "Accepted"
+		)
+
+		poolNN := types.NamespacedName{Name: poolName, Namespace: appBackendNamespace}
+		httpRoute1NN := types.NamespacedName{Name: httpRoute1Name, Namespace: appBackendNamespace}
+		httpRoute2NN := types.NamespacedName{Name: httpRoute2Name, Namespace: appBackendNamespace}
+		gateway1NN := types.NamespacedName{Name: sharedGateway1Name, Namespace: infraNamespace}
+		gateway2NN := types.NamespacedName{Name: sharedGateway2Name, Namespace: infraNamespace}
+
+		routeAcceptedCondition := metav1.Condition{
+			Type:   string(gatewayv1.RouteConditionAccepted),
+			Status: metav1.ConditionTrue,
+			Reason: string(gatewayv1.RouteReasonAccepted),
+		}
+		routeResolvedRefsCondition := metav1.Condition{
+			Type:   string(gatewayv1.RouteConditionResolvedRefs),
+			Status: metav1.ConditionTrue,
+			Reason: string(gatewayv1.RouteReasonResolvedRefs),
+		}
+
+		t.Logf("Waiting for HTTPRoute %s to be Accepted and have ResolvedRefs by Gateway %s", httpRoute1NN.String(), gateway1NN.String())
```

Comment:
Similar comment as another test - but would be good to simplify this even further if possible because I'd expect this to be repeated frequently throughout our tests.

Reply:
Done, that is a good point! I will refactor other PRs accordingly after they are merged ( to avoid duplicating code reviews )